core/src/hir/lifetimes.rs

```diff
@@ -5,14 +5,25 @@ use super::IdentBuf;
 use crate::ast;
 use smallvec::{smallvec, SmallVec};
 
-const EXPECTED_LIFETIMES: usize = 4;
+const INLINE_NUM_LIFETIMES: usize = 4;
```

Comment:
I'd drop a comment saying what this is for, since it's only just an optimization but might seem to be something more